applications/tari_base_node/src/builder.rs

```diff
@@ -473,19 +471,42 @@ where
         .set_base_node_public_key(base_node_public_key.clone())
         .await
         .expect("Problem setting local base node public key for transaction service.");
-    let mut oms_handle = wallet_handles
-        .get_handle::<OutputManagerHandle>()
-        .expect("OutputManagerService is not registered");
-    oms_handle
-        .set_base_node_public_key(base_node_public_key)
-        .await
-        .expect("Problem setting local base node public key for output manager service.");
     // Start the Output Manager UTXO Validation
-    oms_handle
-        .validate_utxos(UtxoValidationRetry::UntilSuccess)
-        .await
-        .expect("Problem starting the Output Manager Service Utxo Valdation process");
-
+    let state_machine = base_node_handles
+        .get_handle::<StateMachineHandle>()
+        .expect("Could not get State Machine handle");
+    let wallet_handles_clone = wallet_handles.clone();
+    task::spawn(async move {
+        let mut status_watch = state_machine.get_status_info_watch();
+        loop {
+            let bootstrapped = match status_watch.recv().await {
+                None => false,
+                Some(s) => match s {
+                    StatusInfo::Listening(li) => li.is_bootstrapped(),
+                    _ => false,
+                },
+            };
+            if bootstrapped {
+                let mut oms_handle = wallet_handles_clone
+                    .get_handle::<OutputManagerHandle>()
+                    .expect("OutputManagerService is not registered");
+                oms_handle
+                    .set_base_node_public_key(base_node_public_key)
+                    .await
+                    .expect("Problem setting local base node public key for output manager service.");
+                oms_handle
+                    .validate_utxos(UtxoValidationRetry::UntilSuccess)
+                    .await
+                    .expect("Problem starting the Output Manager Service Utxo Valdation process");
+                break;
+            }
+            debug!(
+                target: LOG_TARGET,
+                "UTXO validation process not started while busy with initial sync.",
+            );
+            tokio::time::delay_for(Duration::from_millis(30_000)).await;
```

Comment:
`delay_for` is imported in builder.rs. 
```suggestion
            delay_for(Duration::from_secs(30)).await;
```